test/integration/dataplane_api_test.go

```diff
@@ -13,32 +13,75 @@ import (
 	"net"
 	"net/http"
 	"os"
+	"path/filepath"
 	"testing"
 	"time"
 
 	"github.com/go-resty/resty/v2"
 	"github.com/nginx/agent/v3/api/http/dataplane"
-	"github.com/nginx/agent/v3/test/utils"
+	"github.com/nginx/agent/v3/test"
+	"github.com/nginx/agent/v3/test/config"
+
+	"github.com/nginx/agent/v3/test/mock"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/network"
 )
 
 var (
-	apiHost string
-	apiPort string
+	apiHost                      string
+	apiPort                      string
+	instancesURL                 string
+	mockManagementPlaneContainer testcontainers.Container
+	mockManagementPlaneAddress   string
 )
 
 func setupTest(tb testing.TB) func(tb testing.TB) {
 	tb.Helper()
 	var container testcontainers.Container
 	ctx := context.TODO()
 
+	tmpDir := os.TempDir()
+	configDir := filepath.Join(tmpDir, "config")
```

Comment:
Why not:
```suggestion
	tmpDir, err := os.MkdirTemp("", "config")
```
Was it just because we want to use the helper function for actually creating the file and directory?